### PR TITLE
fix creating departments with users

### PR DESCRIPTION
**Pull request by BelehovEgor**
And update find and get position endpoint
Added pagination to find department endpoint
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/802

### CODE REVIEW

File: src/CompanyService.Business/Commands/Position/FindPositionsCommand.cs [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
             IPositionRepository repository,
-            IPositionResponseMapper mapper)
+            IPositionInfoMapper mapper)
         {
             _repository = repository;
             _mapper = mapper;
         }
 
-        public List<PositionResponse> Execute(bool includeDeactivated)
+        public FindPositionsResponse Execute(int skipCount, int takeCount, bool includeDeactivated)
```
Comment (MariyaSin) [inline src/CompanyService.Business/Commands/Position/FindPositionsCommand.cs:23]: You can use FindResultResponse<> from kernel

**Review by MariyaSin — APPROVED** [review]

File: src/CompanyService.Business/Commands/Position/FindPositionsCommand.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/CompanyService.Business/Commands/Position/FindPositionsCommand.cs]: fixed

File: src/CompanyService.Models.Dto/Responses/FindPositionsResponse.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+using System.Collections.Generic;
+
+namespace LT.DigitalOffice.CompanyService.Models.Dto.Responses
+{
+    public record FindPositionsResponse
+    {
+        public int TotalCount { get; set; }
+        public List<PositionInfo> Positions { get; set; } = new();
+        public List<string> Errors { get; set; } = new();
+    }
```
Comment (djino97) [inline src/CompanyService.Models.Dto/Responses/FindPositionsResponse.cs]: May be use find model from Kernel?

**Review by djino97 — APPROVED** [review]

**Review by TIove — APPROVED** [review]

File: src/CompanyService.Models.Dto/Responses/FindPositionsResponse.cs (hunk 2 above)
Comment (BelehovEgor) [reply src/CompanyService.Models.Dto/Responses/FindPositionsResponse.cs]: fixed

**Review by TIove — APPROVED** [review]

File: src/CompanyService.Data/OfficeUserRepository.cs [hunk 3]
```diff
@@ -26,36 +27,24 @@ public void Add(DbOfficeUser user)
...
         {
-            var officeUser = _provider.OfficeUsers.FirstOrDefault(x => x.UserId == userId);
+            return _provider.OfficeUsers.Where(x => userIds.Contains(x.UserId)).Include(x => x.Office).ToList();
+        }
 
-            if (officeUser != null)
+        public void Remove(Guid userId)
+        {
+            DbOfficeUser user = _provider.OfficeUsers.FirstOrDefault(u => u.UserId == userId && u.IsActive);
+
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/CompanyService.Data/OfficeUserRepository.cs:42]: i don't want throw exc here, I just want disactivated record if it exists

File: src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs [hunk 4]
```diff
@@ -69,37 +61,82 @@ private List<UserData> GetUsers(List<Guid> usersIds, List<string> errors)
             return users;
         }
 
-        public DepartmentsResponse Execute()
+        private List<ImageData> GetImages(List<Guid> imageIds, List<string> errors)
         {
-            DepartmentsResponse response = new ();
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs:74]: fixed

**Pull request merged**